docs/TransparentLogging.md

```diff
@@ -0,0 +1,280 @@
+rather than the blobs themselves.
+
+This approach may also be useful if there are situations where full public
+access to the logged data is not appropriate.
+
+ - A Binary Transparency log might include pre-release packages that should not
+   (yet) be available to the general public.
+ - A Pastebin log might end up containing content that is not legal to
+   distribute in some jurisdiction.
+
+## Admission Control
+
+The general idea of transparency can be broken down into two constituent parts,
+which we'll call **read-transparency** and **write-transparency**:
+ - A read-transparent Log allows any external client to read the contents of the
+   Log, and to request cryptographic proofs of the append-only properties
+   (inclusion proofs and consistency proofs).
+ - A write-transparent Log allows any external client to submit new entries to
+   the Log, and receive either an inclusion proof for the entry or a signed
+   promise that it will be incorporated in future.
+
+Allowing write-transparency is vital for encouraging an ecosystem to grow up
+around the Log(s), and helps to reassure external (read-transparency) users that
+no "filtering" has been applied to content before it even reaches the log.
+
+However, it's worth making clear: a write-transparent Log allows arbitrary
+people on the Internet to write data into your append-only Log – and that **data
+can't be removed<sup>[3](#note-3)</sup> without destroying the Log**.
+
+That's a sufficiently terrifying prospect to motivate having strict **admission
+criteria**: checks that submitted content has to pass before being included in
+the Log.  (However, this isn't specific to write-transparent Logs; even if
+submissions are restricted to a whitelist of clients, the admission criteria
+still need to be detailed.)
+
+Examples:
+
+ - For Certificate Transparency, each submission has to take the form of an
+   X.509 certificate in ASN.1 DER format, together with a chain of CA
+   certificates that leads to one of a set of accepted root certificates.
+ - For Binary Transparency, each submission might need to be of the correct
+   layout/format to allow installation on a device, and might need to be signed
+   by one of a set of acceptable package signing keys.
+ - For a Pastebin Log, the admission criteria might merely be a size limit, or a
+   UTF-8 encoding correctness check.
+
+These examples illustrate that good admission criteria typically include two key aspects:
+ - Structure: The submissions have to be of the correct format to be of interest
+   (so that the submissions match the subject of the Log), and that format has
+   to be machine-checkable (but need not be strict – see below).
+ - Authentication: The submissions are signed by one of a limited number of
+   private keys, so the signature can be checked with the corresponding public
+   key.  The Log is thus implicitly
+     - assuming that the holders of those private keys will not sign excessive
```

Comment:
I'm not sure this is a good position for Log operators to take; the key holders are going to do whatever their business/hobby/rogue employee/hacker/whatever drives them to do, regardless of the desires of Log operators[1].
It's also likely that, even at a modest rate of generation, over time a significant body of loggable things will accrue (and indeed, the logs themselves will serve as a convenient place to find all such things), so a new log operator turning up  should probably not make too many assumptions about the rate of loggable things being produced vs. the log throughput figures.

Or, in other words, they can make all the assumptions they like, but they'd probably be wise to have a technical enforcement of them wherever possible :)

[1] OTOH, in a functioning ecosystem there probably _is_ a vested interest in the entities producing the loggable things not destroying all the logs by swamping, but that's not the only

Reply:
Reworded and reordered.